### PR TITLE
Add segment input

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Review by thenav56 — APPROVED** [review]

File: components/Input/SegmentInput/Option/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    readOnly: PropTypes.bool,
+};
+
+const defaultProps = {
+    error: '',
+    disabled: false,
+    readOnly: false,
+};
+
+export default class Input extends React.PureComponent {
```
Comment (frozenhelium) [inline components/Input/SegmentInput/Option/index.js]: Fix component name
Input -> Option

File: components/Input/SegmentInput/Option/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    getClassName = () => {
+        const {
+            disabled,
+            error,
+            readOnly,
+            checked,
+        } = this.props;
+
+        const classNames = [
+            'segment-name',
```
Comment (frozenhelium) [inline components/Input/SegmentInput/Option/index.js]: Fix global className
segment-name -> segment-option
or, something suitable

**Review by frozenhelium — CHANGES_REQUESTED** [review]

File: components/Input/SegmentInput/Option/index.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+        const classNames = this.getClassName();
+
+        return (
+            <label
+                htmlFor={this.inputId}
+                className={classNames}
+            >
+                <input
+                    className={styles.segmentButtonInput}
```
Comment (frozenhelium) [inline components/Input/SegmentInput/Option/index.js]: Maybe add a global className for input

File: components/Input/SegmentInput/Option/styles.scss [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    &.disabled {
+        pointer-events: none;
+    }
+
+    &.read-only {
+        pointer-events: none;
+    }
+
+    &:hover {
```
Comment (frozenhelium) [inline components/Input/SegmentInput/Option/styles.scss]: should be inside 
:not(.disabled) selector

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**